### ISSUE TITLE
[bug]: Converted diffuser model do not get the same/close image with same prompt and seed

### ISSUE THREAD

**Issue by i3oc9i** [invoke-ai/InvokeAI#2713]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

macOS

### GPU

mps

### VRAM

128G

### What happened?

I'have tried to convert a safetensors model to a diffuser one, then I replay exactly the same prompt, with the same settings, and same seed.

The result image is very far from the safetensors one, I am expecting the same or very close result...

_safetensors image_
![001782 17345909](https://user-images.githubusercontent.com/4766500/219992416-77f4baa9-60fa-4efd-bd1b-9dd87eb9ddf2.png)

_diffuser image_
![001783 17345909](https://user-images.githubusercontent.com/4766500/219992549-4bb26c81-a5be-4380-a721-f0158576d2b9.png)

### Screenshots
Here the log where I convert the kalista safetensors model into diffuser one, then I run the same prompt ...

```
* Initializing, be patient...
>> Initialization file /Users/ivano/Code/Ai/@Stuffs/invokeai.models/invokeai.init found. Loading...
>> Internet connectivity is True
>> InvokeAI, version 2.3.0
>> InvokeAI runtime directory is "/Users/ivano/Code/Ai/@Stuffs/invokeai.models"
>> GFPGAN Initialized
>> CodeFormer Initialized
>> ESRGAN Initialized
>> Using device_type mps
>> xformers not installed
>> Current VRAM usage:  0.00G
>> Loading midjourney from /Users/ivano/junk/sd/midjourney-v4.ckpt
>> Scanning Model: midjourney
>> Model scanned ok!
>> Loading midjourney from /Users/ivano/junk/sd/midjourney-v4.ckpt
   | Forcing garbage collection prior to loading new model
   | LatentDiffusion: Running in eps-prediction mode
   | DiffusionWrapper has 859.52 M params.
   | Making attention of type 'vanilla' with 512 in_channels
   | Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
   | Making attention of type 'vanilla' with 512 in_channels
   | Using more accurate float32 precision
   | Loading VAE weights from: /Users/ivano/Code/Ai/@Stuffs/invokeai.models/models/ldm/stable-diffusion-v1/vae-ft-mse-840000-ema-pruned.ckpt
>> Model loaded in 5.15s
>> Model loaded in 5.27s
>> Textual inversions available:
>> Setting Sampler to k_lms

* Initialization done! Awaiting your command (-h for help, 'q' to quit)
(midjourney) invoke> !switch kalista
>> Current VRAM usage:  0.00G
>> Offloading midjourney to CPU
>> Loading kalista from /Users/ivano/junk/sd/kalista.ckpt
>> Scanning Model: kalista
>> Model scanned ok!
>> Loading kalista from /Users/ivano/junk/sd/kalista.ckpt
   | Forcing garbage collection prior to loading new model
   | LatentDiffusion: Running in eps-prediction mode
   | DiffusionWrapper has 859.52 M params.
   | Making attention of type 'vanilla' with 512 in_channels
   | Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
   | Making attention of type 'vanilla' with 512 in_channels
   | Using more accurate float32 precision
   | Loading VAE weights from: /Users/ivano/Code/Ai/@Stuffs/invokeai.models/models/ldm/stable-diffusion-v1/vae-ft-mse-840000-ema-pruned.ckpt
>> Model loaded in 5.43s
>> Model loaded in 5.61s
>> Textual inversions available:
>> Setting Sampler to k_lms
(kalista) invoke> photographic, colour photo, photogenic, rule of thirds, dramatic lighting, fine art dark beauty, detailed face, pale skin, detailed nose, young est woman, (nursing bra)+++, (pointing nipples)+, (bob cuts blond hair)+++ , red lips, (some little flowers come out of the lips of the mouth)++, flower bondage, green eyes, (eyeliner)+, collar or choker, smirk, (dark background)+++, by Paolo roversi [border, frame, vignette, black&white, busty, big breasts, hands]" -s 25 -S 17345909 -W 512 -H 512 -C 7.5 -A k_euler_a
objc[49033]: Class CaptureDelegate is implemented in both /Users/ivano/Code/Ai/@Stuffs/invokeai.models/.venv.old/lib/python3.10/site-packages/cv2/cv2.abi3.so (0x14be124d0) and /opt/homebrew/Cellar/opencv/4.7.0_1/lib/libopencv_videoio.4.7.0.dylib (0x3518f8880). One of the two will be used. Which one is undefined.
objc[49033]: Class CVWindow is implemented in both /Users/ivano/Code/Ai/@Stuffs/invokeai.models/.venv.old/lib/python3.10/site-packages/cv2/cv2.abi3.so (0x14be12520) and /opt/homebrew/Cellar/opencv/4.7.0_1/lib/libopencv_highgui.4.7.0.dylib (0x30f6a0b10). One of the two will be used. Which one is undefined.
objc[49033]: Class CVView is implemented in both /Users/ivano/Code/Ai/@Stuffs/invokeai.models/.venv.old/lib/python3.10/site-packages/cv2/cv2.abi3.so (0x14be12548) and /opt/homebrew/Cellar/opencv/4.7.0_1/lib/libopencv_highgui.4.7.0.dylib (0x30f6a0b38). One of the two will be used. Which one is undefined.
objc[49033]: Class CVSlider is implemented in both /Users/ivano/Code/Ai/@Stuffs/invokeai.models/.venv.old/lib/python3.10/site-packages/cv2/cv2.abi3.so (0x14be12570) and /opt/homebrew/Cellar/opencv/4.7.0_1/lib/libopencv_highgui.4.7.0.dylib (0x30f6a0b60). One of the two will be used. Which one is undefined.
>> Patchmatch initialized
>> Setting Sampler to k_euler_a
>> Prompt is 8 token(s) too long and has been truncated
/Users/ivano/Code/Ai/invokeai/ldm/modules/embedding_manager.py:146: UserWarning: The operator 'aten::nonzero' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
  placeholder_rows, placeholder_cols = torch.where(
Generating:   0%|                                                                                                                                                                                                                                                                                                                          | 0/1 [00:00<?, ?it/s]>> Ksampler using karras noise schedule (steps < 30)
>> Sampling with k_euler_ancestral starting at step 0 of 25 (25 new sampling steps)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 25/25 [00:10<00:00,  2.29it/s]
Generating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:11<00:00, 11.46s/it]

>> Usage stats:
>>   1 image(s) generated in 12.06s
Outputs:
[198] /Users/ivano/Code/Ai/@Stuffs/images/@invokeai/001782.17345909.png: photographic, colour photo, photogenic, rule of thirds, dramatic lighting, fine art dark beauty, detailed face, pale skin, detailed nose, young est woman, (nursing bra)+++, (pointing nipples)+, (bob cuts blond hair)+++ , red lips, (some little flowers come out of the lips of the mouth)++, flower bondage, green eyes, (eyeliner)+, collar or choker, smirk, (dark background)+++, by Paolo roversi [border, frame, vignette, black&white, busty, big breasts, hands]" -s 25 -S 17345909 -W 512 -H 512 -C 7.5 -A k_euler_a


(kalista) invoke> !switch midjourney
>> Current VRAM usage:  0.00G
>> Offloading kalista to CPU
>> Retrieving model midjourney from system RAM cache
>> Textual inversions available:
>> Setting Sampler to k_euler_a

(midjourney) invoke> !optimize_model kalista
Replace this model's VAE with "stabilityai/sd-vae-ft-mse"? [y/N]: y
>> Optimizing kalista (30-60s)
  | Loading diffusers VAE from stabilityai/sd-vae-ft-mse
  | Using more accurate float32 precision
  | global_step key not found in model
  | Checkpoint /Users/ivano/junk/sd/kalista.ckpt has both EMA and non-EMA weights.
  | Extracting EMA weights (usually better for inference)
  | Using external VAE specified in config
  | Success. Optimized model is now located at /Users/ivano/Code/Ai/@Stuffs/invokeai.models/models/converted_ckpts/kalista
  | Writing new config file entry for kalista
>> Offloading kalista to CPU
>> Conversion succeeded
Load optimized model kalista? [Y/n]:
>> Current VRAM usage:  0.00G
>> Offloading midjourney to CPU
>> Loading diffusers model from /Users/ivano/Code/Ai/@Stuffs/invokeai.models/models/converted_ckpts/kalista
  | Using more accurate float32 precision
  | Calculating sha256 hash of model files
  | sha256 = 05d3e4ea90d48aac70223fba0f9be6a2a7b83280b45c2a141ec2b5e75a1aeebd (15 files hashed in 3.57s)
  | Default image dimensions = 512 x 512
>> Model loaded in 4.83s
>> Textual inversions available:
>> Setting Sampler to k_euler_a (EulerAncestralDiscreteScheduler)
Delete the original .ckpt file at /Users/ivano/junk/sd/kalista.ckpt? [y/N]:
(kalista) invoke> !switch kalista
(kalista) invoke> photographic, colour photo, photogenic, rule of thirds, dramatic lighting, fine art dark beauty, detailed face, pale skin, detailed nose, young est woman, (nursing bra)+++, (pointing nipples)+, (bob cuts blond hair)+++ , red lips, (some little flowers come out of the lips of the mouth)++, flower bondage, green eyes, (eyeliner)+, collar or choker, smirk, (dark background)+++, by Paolo roversi [border, frame, vignette, black&white, busty, big breasts, hands]" -s 25 -S 17345909 -W 512 -H 512 -C 7.5 -A k_euler_a
>> Prompt is 8 token(s) too long and has been truncated
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 25/25 [00:09<00:00,  2.68it/s]
Generating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:09<00:00,  9.78s/it]

>> Usage stats:
>>   1 image(s) generated in 10.02s
Outputs:
[199] /Users/ivano/Code/Ai/@Stuffs/images/@invokeai/001783.17345909.png: photographic, colour photo, photogenic, rule of thirds, dramatic lighting, fine art dark beauty, detailed face, pale skin, detailed nose, young est woman, (nursing bra)+++, (pointing nipples)+, (bob cuts blond hair)+++ , red lips, (some little flowers come out of the lips of the mouth)++, flower bondage, green eyes, (eyeliner)+, collar or choker, smirk, (dark background)+++, by Paolo roversi [border, frame, vignette, black&white, busty, big breasts, hands]" -s 25 -S 17345909 -W 512 -H 512 -C 7.5 -A k_euler_a

(kalista) invoke>
```

### Additional context
Commit: 7eafcd47 (HEAD -> main, origin/main, origin/HEAD) [WebUI] Bug Fixes (#2728)

_safetensors model_
```
kalista:
  description: Kalista [photo]
  config: configs/stable-diffusion/v1-inference.yaml
  weights: /Users/ivano/junk/sd/kalista.ckpt
  vae: models/ldm/stable-diffusion-v1/vae-ft-mse-840000-ema-pruned.ckpt
  width: 512
  height: 512
  format: ckpt
```
  
 _diffuser model_
```
kalista:
  path: /Users/ivano/Code/Ai/@Stuffs/invokeai.models/models/converted_ckpts/kalista
  description: Kalista [photo]
  format: diffusers
```

### Contact Details

_No response_

**Comment by i3oc9i** [comment]
Indeed, I did some other test with other models, some are well converted, but some are not, for example this one is not  https://civitai.com/models/4825/kalista

FYI, the VAE used with the Kalista.ckpt model was the usal vae-ft-mse-840000-ema-pruned.ckpt), and 
And i say Yes to this question, Replace this model's VAE with "stabilityai/sd-vae-ft-mse"? [y/N]: y
If I answer No to the VAE question, then the image I get for the second one (diffuser model)  is less saturated/fade] 

![001783 e7e9d681 17345909](https://user-images.githubusercontent.com/4766500/219941040-c43f8eb4-241d-441d-bc2c-2c9f31a9f066.png)
![001786 d3742a00 17345909](https://user-images.githubusercontent.com/4766500/219941043-b9d58a94-3baa-4d2d-902e-b1ebd3070225.png)

**Comment by i3oc9i** [comment]
this is another example using https://civitai.com/models/7279/colorful (ver 1.4)

original safetensors
![001784 835051b8 394878809](https://user-images.githubusercontent.com/4766500/219953202-ea9d3f88-5bd3-46f2-a40e-49b2ba074e66.png)

diffuser one
![001785 75f941d2 394878809](https://user-images.githubusercontent.com/4766500/219953218-e0d73ff6-07f4-4534-b8da-6dc6e8bb64d1.png)

**Comment by i3oc9i** [comment]
I take the time to do some more test, when I use simple prompts with no more than one attention, then the images are the same between diffuser model and original one, but if the prompt contains severals attentions then they are different

**Comment by i3oc9i** [comment]
QGO-MK8 is probably a broken model

**Comment by i3oc9i** [comment]
FYI, the VAE related to the Kalista.ckpt model is the usual vae-ft-mse-840000-ema-pruned.ckpt) 

If I answer `No` to the VAE question during the optimisation process, then I get a  desaturated/fade image corresponding to the one of `diffuser model` reported in the issue.

**Comment by lstein** [comment]
The issue appears to be related to `xformers`, which is enabled for `diffusers` models and disabled for legacy checkpoints. If you launch with `--no-xformers` then the images from the converted and original models are almost the same. They're not identical, but pretty close.

**Comment by i3oc9i** [comment]
**good point for PC/Linux users**, but I'm on MacOs so _xformers_ are not installed.. :) 
( _I have anyway started the invoke how you suggested with the_ `--no-xformers`)

I was getting crazy about this issue, but I think I get the point. I did more test with the `kalista` (from civitai) model.

>  banana sushi -s 29 -S 123456789 -W 512 -H 512 -C 7.5 -A k_lms 
>  banana sushi -s 30 -S 123456789 -W 512 -H 512 -C 7.5 -A k_lms [**OK**]
> 
> banana sushi -s 29 -S 123456789 -W 512 -H 512 -C 7.5 -A k_euler_a
> banana sushi -s 30 -S 123456789 -W 512 -H 512 -C 7.5 -A k_euler_a [**OK**]
> 
> banana sushi -s 29 -S 123456789 -W 512 -H 512 -C 7.5 -A k_heun
> banana sushi -s 30 -S 123456789 -W 512 -H 512 -C 7.5 -A k_heun [**OK**]

 So **Keras** diffusers are not the same under 30 steps..., but they match at 30 steps and over
 
 _ddim_ and _plms_ diffusers are still different at 100 steps
